internal/bminventory/inventory_v2_handlers.go

```diff
@@ -671,6 +671,29 @@ func (b *bareMetalInventory) V2SetIgnoredValidations(ctx context.Context, params
 	return installer.NewV2SetIgnoredValidationsCreated().WithPayload(&ignoredValidations)
 }
 
+func (b *bareMetalInventory) V2GetClusterInstallerUISettings(ctx context.Context, params installer.V2GetClusterInstallerUISettingsParams) middleware.Responder {
+	cluster, err := b.getCluster(ctx, params.ClusterID.String())
+	if err != nil {
+		err = errors.Wrapf(err, "failed to fetch cluster %s to get UI settings", *cluster.ID)
+		return installer.NewV2GetClusterInstallerUISettingsInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	return installer.NewV2GetClusterInstallerUISettingsOK().WithPayload(cluster.UISettings)
+}
+
+func (b *bareMetalInventory) V2UpdateClusterInstallerUISettings(ctx context.Context, params installer.V2UpdateClusterInstallerUISettingsParams) middleware.Responder {
+	cluster, err := b.getCluster(ctx, params.ClusterID.String())
+	if err != nil {
+		err = errors.Wrapf(err, "failed to fetch cluster %s to get UI settings", *cluster.ID)
+		return installer.NewV2UpdateClusterInstallerUISettingsInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	cluster.UISettings = params.UISettings
+	if err = b.db.Save(cluster).Error; err != nil {
```

Comment:
Is this really the best way to do this? Theoretically this could be single statement in sql. Do we have a way to do that?

Reply:
Alternatively you could start a transaction and do a get `FOR UPDATE`.

The install config overrides endpoint does this but I imagine that's just because it also needs to do validation.